src/jit/compiler.cpp

```diff
@@ -4191,6 +4191,16 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
 
     fgImport();
 
+    if (IsTargetAbi(CORINFO_CORERT_ABI) && hasFatPointer())
+    {
+        if (fgCheapPredsValid)
+        {
```

Comment:
Also add a comment explaining what fgCheapPredsValid has to do with any of this.